Rename Typ.join to Typ.meet to be more consistent with other papers

7h3kk1d opened the pull request:
type system's 'join' operation was actually computing the meet
(greatest lower bound) since Unknown represents the highest element in the precision ordering.

- Rename Typ.join → Typ.meet
- Rename Typ.join_all → Typ.meet_all
- Rename join_type_provenance → meet_type_provenance
- Update comment to reference 2019 Hazelnut live paper join

Fixes #2018

7h3kk1d committed:
Rename Typ.join to Typ.meet to be more consistent with other papers

 type system's 'join' operation was actually computing the meet
(greatest lower bound) since Unknown represents the top element in
the precision ordering. This change corrects the naming to align
with standard semi-lattice theory and clarifies the lattice structure.

- Rename Typ.join → Typ.meet
- Rename Typ.join_all → Typ.meet_all
- Rename join_type_provenance → meet_type_provenance
- Update comment to reference 2019 Hazelnut live paper join

7h3kk1d committed:
Finish the remaining conversions of typ join to meet

7h3kk1d committed:
Merge remote-tracking branch 'origin/dev' into join-to-meet-rename

cyrus- commented on test/statics/Test_Statics_Properties.re:
```diff
@@ -11,7 +11,7 @@ let qcheck_statics_does_not_crash =
     | exception (Failure(f) as e) =>
       switch (f) {
-      | "Type join of ap" => true // TODO https://github.com/hazelgrove/hazel/issues/1459
+      | "Type meet of ap" => true // TODO https://github.com/hazelgrove/hazel/issues/1459
```
the linked issue here is closed

7h3kk1d replied:
Resolved in https://github.com/hazelgrove/hazel/pull/2027/commits/eceb0ade8cfa50b97c383a8fb19a17a4d6de6719

cyrus- reviewed (dismissed):
minor comment, go ahead and merge once you're ready

7h3kk1d committed:
Merge remote-tracking branch 'origin/dev' into join-to-meet-rename

7h3kk1d committed:
Update tests to reflect issue 1459 being fixed

- Remove "Type meet of ap" from known failures in Test_Elaboration.re and Test_Statics_Properties.re
- Convert skip_known_bug for Type meet of ap in Test_Statics_Types.re to a proper test case checking for ParseFailure error, addressing issue #1459
